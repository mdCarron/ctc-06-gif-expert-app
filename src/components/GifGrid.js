import React, { useEffect } from "react";
import PropTypes from "prop-types";

const GifGrid = ({ categoria }) => {
  useEffect(() => {
    getGifs();
  }, []);

  const getGifs = async () => {
    const url =
      "http://api.giphy.com/v1/gifs/search?q=nes&limit=10&api_key=pZF5taT2C8y5Jk6AOVAZpZhF7B0b4j17";
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map((gif) => {
      return {
        id: gif.id,
        title: gif.title,
        url: gif.images.downsized_medium.url,
      };
    });
    console.log(gifs);
  };

  return (
    <div>
      <h3>{categoria}</h3>
    </div>
  );
};

GifGrid.propTypes = {};

export default GifGrid;
