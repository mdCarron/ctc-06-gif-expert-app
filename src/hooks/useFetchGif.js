import { useState } from "react";

export const useFetchGif = () => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  return state;
};
