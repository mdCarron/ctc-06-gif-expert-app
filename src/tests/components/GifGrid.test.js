import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import GifGrid from "../../components/GifGrid";
import { useFetchGif } from "../../hooks/useFetchGif";
jest.mock("../../hooks/useFetchGif");

describe("testing en componente <GifGrid />", () => {
  const categoria = "super nintendo";

  test("should matchear snapshot", () => {
    useFetchGif.mockReturnValue({
      data: [],
      loading: true,
    });

    const wrapper = shallow(<GifGrid categoria={categoria} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("should debe de mostrar items cuando se cargan las imagenes con useFetchGif", () => {
    const gifs = [
      {
        id: "ABC",
        title: "MockTitle",
        url: "https://localhost/img.gif",
      },
    ];

    useFetchGif.mockReturnValue({
      data: gifs,
      loading: false,
    });

    const wrapper = shallow(<GifGrid categoria={categoria} />);

    expect(wrapper).toMatchSnapshot();
  });
});
