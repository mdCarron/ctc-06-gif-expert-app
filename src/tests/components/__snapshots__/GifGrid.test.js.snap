// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing en componente <GifGrid /> should debe de mostrar items cuando se cargan las imagenes con useFetchGif 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeft"
  >
    super nintendo
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="1"
      key="1"
      title="MockTitle"
      url="https://localhost/img.gif"
    />
    <GifGridItem
      id="2"
      key="2"
      title="MockTitle"
      url="https://localhost/img.gif"
    />
  </div>
</Fragment>
`;

exports[`testing en componente <GifGrid /> should matchear snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInLeft"
  >
    super nintendo
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    generando gifs...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
